### PR TITLE
chore: update deps (vello, thiserror)

### PR DESCRIPTION
**Pull request by simbleau**
New version `0.4.0`: 
- Update to vello 0.3
- thiserror to 2.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -68,6 +77,8 @@ This release has an [MSRV][] of 1.75.
 [#42]: https://github.com/linebender/velato/pull/42
 
 [Unreleased]: https://github.com/linebender/velato/compare/v0.3.0...HEAD
+[0.4.0]: https://github.com/linebender/velato/compare/v0.3.1...v0.4.0
+[0.3.1]: https://github.com/linebender/velato/compare/v0.3.0...v0.3.1
```
Comment (simbleau) [inline CHANGELOG.md:81]: Forgot this link on the last release, including it here.

Comment (waywardmonkeys) [reply CHANGELOG.md:81]: That's why you should do a release as a separate PR and let someone else review it. Might've caught it.

File: examples/with_winit/src/lib.rs [hunk 2]
```diff
@@ -376,9 +377,14 @@ fn run(
...
                             .expect("failed to get surface texture");
+                        // Note: we don't run the async/"robust" pipeline, as
+                        // it requires more async wiring for the readback. See
+                        // [#gpu > async on wasm](https://xi.zulipchat.com/#narrow/stream/197075-gpu/topic/async.20on.20wasm)
+                        #[allow(deprecated)]
+                        // #[expect(deprecated, reason = "This deprecation is not targeted at us.")] // Our MSRV is too low to use `expect`
                         #[cfg(not(target_arch = "wasm32"))]
                         {
-                            scene_complexity = vello::block_on_wgpu(
+                            scene_complexity = vello::util::block_on_wgpu(
```
Comment (DJMcNab) [inline examples/with_winit/src/lib.rs:387]: I'd suggest just removing the async pipeline entirely.

Comment (simbleau) [reply examples/with_winit/src/lib.rs:387]: Would there be a better approach to take entirely?

The examples are copied from vello, but done so a while back. They don't have parity either, some changes were made to play lottie files obviously.

Would it be better to just remove all examples and just have a very simple wasm/native example without all the diagnostics and head scratching?

Comment (simbleau) [reply examples/with_winit/src/lib.rs:387]: If so, I'd be fine pushing this through and then making a follow up PR to gut the examples with a simpler one.

Comment (DJMcNab) [reply examples/with_winit/src/lib.rs:387]: Removing that pipeline doesn't matter that much in the grand scheme of things - this example is more of a demo than an actionable example.

**Review by DJMcNab — APPROVED** [review]
Approving to unblock. I'd still prefer the async pipeline usage be removed, as that API is no longer bound by our upstream semver guarantees.

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,15 @@ Subheadings to categorize changes are `added, changed, deprecated, removed, fixe
 
 This release has an [MSRV][] of 1.75.
 
+## [0.4.0]  - 2024-11-11
```
Comment (DJMcNab) [inline CHANGELOG.md:15]: Why is this release date here? This hasn't been released.

Comment (simbleau) [reply CHANGELOG.md:15]: Oh, I thought I did.

**Pull request merged**